### PR TITLE
Add tests for ShouldIgnoreVnodeType

### PR DESCRIPTION
**Pull request by jeschu1**
Adds tests for ShouldIgnoreVnodeType

### CODE REVIEW

File: ProjFS.Mac/PrjFSKextTests/KextLogMock.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef KextLogMock_h
+#define KextLogMock_h
+
+inline void KextLog_Error(const char* fmt, ...) {}
```
Comment (jeschu1) [inline ProjFS.Mac/PrjFSKextTests/KextLogMock.h]: We could do something fancier for testing logging in the future. Not something to address now.

File: ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm [hunk 2]
```diff
@@ -32,4 +32,18 @@ - (void)testFileFlagsBitIsSet {
...
+    XCTAssertTrue(ShouldIgnoreVnodeType(VCHR, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VSOCK, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VFIFO, NULL));
+    XCTAssertTrue(ShouldIgnoreVnodeType(VBAD, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VREG, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VDIR, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VLNK, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VSTR, NULL));
+    XCTAssertFalse(ShouldIgnoreVnodeType(VCPLX, NULL));
+}
```
Comment (wilbaker) [inline ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: Also add a test that `ShouldIgnoreVnodeType` returns false for an unknown type?

Comment (jeschu1) [reply ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: ShouldIgnoreVnodeType has takes in 'enum vtype' so I have to send that in.  I have a test case for each vtype.  Did you have something else in mind @wilbaker

Comment (wilbaker) [reply ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm]: I was thinking something like:

```
// 10000 isn't found in the vtype enum
XCTAssertFalse(ShouldIgnoreVnodeType(static_cast<vtype>(10000), NULL));
```

To test this part of `ShouldIgnoreVnodeType`:

```
    default:
        KextLog_FileInfo(vnode, "vnode with unknown type %d encountered", vnodeType);
        return false;
```

**Pull request merged**